### PR TITLE
[CDAP-12849] add graph interaction with legends and app types

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-12849

### Additions:
- Legends will toggle the graph
- App Type selection will modify the data being displayed on the graph
- Window resize will adjust graph width

### Approach:
Currently any modification will remove existing graph and render a new one. It's a suboptimal approach, but making the graph dynamic was much more complicated than I initially anticipated, so I reverted to this approach. Will revisit this updating function later.

### CODE REVIEW

File: cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js [hunk 1]
```diff
@@ -15,71 +15,50 @@
...
-
-        // Render Graph
-        let containerElem = document.getElementById(RUNS_GRAPH_CONTAINER);
-
-        let width = containerElem.offsetWidth,
-            height = 230;
-        renderGraph('#runs-graph', width, height, this.state.data);
-      });
+  componentWillReceiveProps(nextProps) {
+    this.renderGraph(nextProps);
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js:48]: Might want to optimize here if the props have actually changed. If not we could skip a re-render. Not the best optimization but anything we can do not re-render a complex graph might ease out the paint cycle.

File: cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js [hunk 2]
```diff
@@ -15,71 +15,50 @@
...
+    if (this.windowResize$ && this.windowResize$.unsubscribe) {
+      this.windowResize$.unsubscribe();
+    }
+  }
+
+  renderGraph(props = this.props) {
+    let containerElem = document.getElementById(RUNS_GRAPH_CONTAINER);
+
+    let width = containerElem.offsetWidth,
+        height = 230;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js]: nit: May be define as constant at the top?

**Review by ajainarayanan — COMMENTED** [review]
Just a few minor comments. Otherwise code LGTM 👍

File: cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js:48]: So the props being passed here is already limited, and any update to the props will require a change. There shouldn't be an update to the props where the data is not changed.

**Comment by ajainarayanan** [comment]
👍 LGTM

**Pull request merged**